src/librustc/traits/select.rs

```diff
@@ -3066,11 +3095,26 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
         })
     }
 
+    fn confirm_trait_alias_candidate(
+        &mut self,
+        obligation: &TraitObligation<'tcx>,
+    ) -> VtableObjectData<'tcx, PredicateObligation<'tcx>> {
+        debug!("confirm_trait_alias_candidate({:?})", obligation);
+
+        todo
```

Comment:
So, what you need to do here is really pretty simple. You're kind of doing a simplified version of what an impl does -- the difference is that impls have types you have to match against, but trait aliases don't.

To start, we would deal with any higher-ranked regions. This starts by creating a snapshot, like so:

https://github.com/rust-lang/rust/blob/5ea8eb55cd9f4547b332f43c9f723de30187c223/src/librustc/traits/select.rs#L2899

Then you would replace the higher-ranked regions with placeholders:

https://github.com/rust-lang/rust/blob/5ea8eb55cd9f4547b332f43c9f723de30187c223/src/librustc/traits/select.rs#L3492-L3493

This gives you back a `skol_obligation` -- basically a version of your obligation where any higher-ranked regions (e.g., the `'a` in `for<'a>`) have been replaced with "placeholders" meant to represent "any possible region". From this you can extract `skol_obligation.predicate.trait_ref` to get a [TraitRef](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/struct.TraitRef.html), which gives you a def-id and substs.

From there, can invoke the existing `impl_or_trait_obligations` helper function, just like `vtable_impl` does:

https://github.com/rust-lang/rust/blob/5ea8eb55cd9f4547b332f43c9f723de30187c223/src/librustc/traits/select.rs#L2930-L2938

And the construct your final result, kind of like this:

https://github.com/rust-lang/rust/blob/5ea8eb55cd9f4547b332f43c9f723de30187c223/src/librustc/traits/select.rs#L2952-L2956